assets/page/SettingsAccount.svelte

```diff
@@ -9,18 +9,30 @@ import {getContext} from 'svelte';
 import {l} from '../js/i18n';
 import {route} from '../store/Route';
 
-const colorSchemeOptions = [['auto', 'Auto'], ['dark', 'Dark'], ['light', 'Light']];
 const user = getContext('user');
 const updateUserOp = user.api.operation('updateUser');
 
+$: calculateColorSchemeOptions(theme);
+
+function calculateColorSchemeOptions(id) {
+  const options = [['auto', 'Auto']];
+  const theme = user.themes[id] || {variants: {}};
+  if (theme.variants.dark) options.push(['dark', 'Dark']);
+  if (theme.variants.light) options.push(['light', 'Light']);
+  colorSchemeOptions = options;
+  colorScheme = 'auto'; 
+}
+
 const themes = Object.keys(user.themes).map(id => {
   let name = user.themes[id].name;
   const colorSchemes = Object.keys(user.themes[id].variants || {}).filter(cs => cs != 'default');
   if (colorSchemes.length) name += ' (' + colorSchemes.sort().join('/') + ')';
   return [id, name];
 });
 
+
 let formEl;
+let colorSchemeOptions;
```

Comment:
Does this work without a default value? It doesn't have to be set to `[]` initially..?

Reply:
It worked without value, but I just added it as you suggested